### PR TITLE
drivers/video: Support spot position setting

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary
Add new control id V4L2_CID_EXPOSURE_METERING_SPOT_POSITION to support spot position setting in spot exposure metering.

## Impact
video driver.

## Testing
Tested with spresense

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

File: drivers/video/isx012.c [hunk 1]
```diff
@@ -2356,6 +2438,31 @@ static int set_clip(uint32_t size,
   return OK;
 }
 
+static int set_spot_position(FAR struct isx012_dev_s *priv, int32_t val)
+{
+  uint16_t w;
+  uint16_t h;
+  uint16_t x = (uint16_t)((val & 0xffff0000) >> 16);
```
Comment (pkarashchenko) [inline drivers/video/isx012.c:2445]: Optional
```suggestion
  uint16_t x = (uint16_t)(val >> 16);
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@SPRESENSE  please rebase your change.

**Comment by SPRESENSE** [comment]
@pkarashchenko @xiaoxiang781216 Thanks for your comments. I updated this PR as suggested.

**Review by jerpelea — APPROVED** [review]

**Comment by acassis** [comment]
@SPRESENSE does the camera support work with NuttX mainline? BTW, I think there is not official documentation to Sony Spresense's chip in our mainline: https://nuttx.apache.org/docs/latest/platforms/arm/index.html could you please submit it?

**Comment by SPRESENSE** [comment]
@acassis 
Yes, the camera works with NuttX mainline.
Thank you for your suggestion. I will try to submit a separate pull request for documentation.

**Pull request merged**